app/bundles/LeadBundle/EventListener/SegmentReportSubscriber.php

```diff
@@ -47,12 +46,18 @@ public function onReportBuilder(ReportBuilderEvent $event): void
                 'label' => 'mautic.lead.report.segment.manually_added',
                 'type'  => 'bool',
             ],
+            'email_domain' => [
+                'label'    => 'mautic.lead.report.segment.domain',
+                'type'     => 'string',
+                'alias'    => 'email_domain',
+                'formula'  => 'CASE WHEN l.email LIKE \'%@%\' THEN SUBSTRING_INDEX(l.email, \'@\', -1) ELSE \'unknown\' END',
```

Comment:
There is already a field `generated_email_domain` in the `leads` table, we could use it instead of this formula to improve performance.